pyrsa/rdm/calc_rdm.py

```diff
@@ -0,0 +1,200 @@
+    elif method == 'correlation':
+        rdm = calc_rdm_correlation(dataset, descriptor)
+    elif method == 'mahalanobis':
+        rdm = calc_rdm_mahalanobis(dataset, descriptor, noise)
+    elif method == 'crossnobis':
+        rdm = calc_rdm_crossnobis(dataset, descriptor, noise)
+    else:
+        raise(NotImplementedError)
+    return rdm
+
+
+def calc_rdm_euclid(dataset, descriptor=None):
+    """
+    calculates an RDM from an input dataset using euclidean distance
+    If multiple instances of the same condition are found in the dataset
+    they are averaged.
+
+        Args:
+            dataset (pyrsa.data.DatasetBase):
+                The dataset the RDM is computed from
+            descriptor (String):
+                obs_descriptor used to define the rows/columns of the RDM
+                defaults to one row/column per row in the dataset
+        Returns:
+            RDMs object with the one RDM
+    """
+    measurements, desc, descriptor = _parse_input(dataset, descriptor)
+    c_matrix = allpairs(np.arange(measurements.shape[0]))
+    diff = np.matmul(c_matrix, measurements)
+    rdm = np.einsum('ij,ij->i', diff, diff) / measurements.shape[1]
+    rdm = RDMs(dissimilarities=np.array([rdm]),
+               dissimilarity_measure='euclidean',
+               descriptors=dataset.descriptors)
+    rdm.pattern_descriptors[descriptor] = desc
+    return rdm
+
+
+def calc_rdm_correlation(dataset, descriptor=None):
+    """
+    calculates an RDM from an input dataset using correlation distance
+    If multiple instances of the same condition are found in the dataset
+    they are averaged.
+
+        Args:
+            dataset (pyrsa.data.DatasetBase):
+                The dataset the RDM is computed from
+            descriptor (String):
+                obs_descriptor used to define the rows/columns of the RDM
+                defaults to one row/column per row in the dataset
+        Returns:
+            RDMs object with the one RDM
+    """
+    measurements, desc, descriptor = _parse_input(dataset, descriptor)
+    rdm = 1 - np.corrcoef(measurements)
+    rdm = RDMs(dissimilarities=np.array([rdm]),
+               dissimilarity_measure='euclidean',
+               descriptors=dataset.descriptors)
+    rdm.pattern_descriptors[descriptor] = desc
+    return rdm
+
+
+def calc_rdm_mahalanobis(dataset, descriptor=None, noise=None):
+    """
+    calculates an RDM from an input dataset using mahalanobis distance
+    If multiple instances of the same condition are found in the dataset
+    they are averaged.
+
+        Args:
+            dataset (pyrsa.data.DatasetBase):
+                The dataset the RDM is computed from
+            descriptor (String):
+                obs_descriptor used to define the rows/columns of the RDM
+                defaults to one row/column per row in the dataset
+            noise (numpy.ndarray):
+                precision matrix used to calculate the RDM
```

Comment:
Could you add something about the expected dimensions of the noise array?

Reply:
I added a line to the help that this is supposed to have dimension n_channel x n_channel and introduced corresponding checks in a new commit